diff -ru SuperLU_MT_3.0-orig/Makefile SuperLU_MT_3.0/Makefile
--- SuperLU_MT_3.0-orig/Makefile	2013-03-20 22:46:57.000000000 +0100
+++ SuperLU_MT_3.0/Makefile	2016-03-06 11:43:44.000000000 +0100
@@ -18,6 +18,10 @@
 all: install lib testing
 
 lib: superlulib tmglib
+	case `uname` in						\
+	Darwin) $(MAKE) -C ./SRC ../lib/libsuperlu_mt_PTHREAD.dylib;;	\
+	*) $(MAKE) -C ./SRC ../lib/libsuperlu_mt_PTHREAD.so;;		\
+	esac
 
 clean: cleanlib cleantesting
 
diff -ru SuperLU_MT_3.0-orig/SRC/Makefile SuperLU_MT_3.0/SRC/Makefile
--- SuperLU_MT_3.0-orig/SRC/Makefile	2015-04-19 20:20:28.000000000 +0200
+++ SuperLU_MT_3.0/SRC/Makefile	2016-03-06 11:43:44.000000000 +0100
@@ -117,6 +117,14 @@
 	$(ARCH) $(ARCHFLAGS) ../lib/$(SUPERLULIB) $(ZLUSRC) $(ALLAUX) $(DZAUX)
 	$(RANLIB) ../lib/$(SUPERLULIB)
 
+../lib/$(SUPERLULIB:.a=.dylib): $(ZLUSRC) $(ALLAUX) $(DZAUX)
+	$(CC) -dynamiclib -shared -undefined dynamic_lookup	\
+	   -compatibility_version 3.0 -current_version 3.0	\
+	$(DLUSRC) $(ALLAUX) $(DZAUX) -o $@
+
+../lib/$(SUPERLULIB:.a=.so): $(ZLUSRC) $(ALLAUX) $(DZAUX)
+	$(CC) -shared $(DLUSRC) $(ALLAUX) $(DZAUX) -o $@
+
 ##################################
 # Do not optimize this routine   #
 ##################################
diff -ru SuperLU_MT_3.0-orig/make.inc SuperLU_MT_3.0/make.inc
--- SuperLU_MT_3.0-orig/make.inc	2015-04-30 21:57:35.000000000 +0200
+++ SuperLU_MT_3.0/make.inc	2016-03-06 11:44:28.000000000 +0100
@@ -22,8 +22,8 @@
 TMGLIB       	= libtmglib$(PLAT).a
 SUPERLULIB   	= libsuperlu_mt$(PLAT).a
 BLASDEF	        = -DUSE_VENDOR_BLAS
-BLASLIB		= -L/usr/lib -lf77blas -latlas
-#BLASLIB = ../lib/libblas$(PLAT).a
+#BLASLIB		= -L/usr/lib -lf77blas -latlas
+BLASLIB		= -lblas
 MATHLIB		= -lm 
 MPLIB		= -lpthread
 
@@ -40,7 +40,7 @@
 # Definitions used by CPP:
 PREDEFS      	= -D__PTHREAD  ## -DDEBUGlevel=0 -DPRNTlevel=0
 #
-CC           	= gcc
+CC           	= gcc -fPIC
 CFLAGS       	= $(PREDEFS) -O3
 # Uncomment the following to use 64-bit integer
 # CFLAGS          += -D_LONGINT
